soroban-sdk/src/env.rs

```diff
@@ -1318,20 +1324,29 @@ thread_local! {
 impl Env {
     /// Create a snapshot file for the currently executing test.
     ///
-    /// Writes the file to the `test_snapshots/{test-name}.json` path.
+    /// Writes the file to the `test_snapshots/{test-name}.N.json` path where
+    /// `N` is incremented for each unique `Env` in the test.
     ///
     /// Use to record the observable behavior of a test, and changes to that
     /// behavior over time. Commit the test snapshot file to version control and
     /// watch for changes in it on contract change, SDK upgrade, protocol
     /// upgrade, and other important events.
     ///
     /// No file will be created if the environment has no meaningful data such
-    /// as stored entries.
+    /// as stored entries or events.
     ///
     /// ### Panics
     ///
     /// If there is any error writing the file.
     pub(crate) fn to_test_snapshot_file(&self) {
+        let snapshot = self.to_snapshot();
+
+        // Don't write a snapshot that has no data in it.
+        if snapshot.ledger.entries().into_iter().count() == 0 && snapshot.events.0.is_empty() {
```

Comment:
Yeah that's fair. I went back and forth on this because I had a ton of tests that wrote almost empty files and that felt not great. But I see your point also.

I'm okay with removing it. I'll do it in another PR. This PR is just moving the logic, the logic existed already.